File: java/src/main/java/gov/doc/ntia/sigmf/ext/global/core/Antenna.java
package gov.doc.ntia.sigmf.ext.global.core;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class Antenna implements Serializable {

    private static final long serialVersionUID = 1L;

    @JsonProperty(value="antenna_spec", required = true)
    protected HardwareSpec antennaSpec;

    @JsonProperty(value="type", required=false)
    protected  String type;


    @JsonProperty(value="frequency_low", required = false)
    protected Double frequencyLow;


    @JsonProperty(value="frequency_high", required = false)
    protected Double frequencyHigh;


    @JsonProperty(value="gain", required=false)
    protected Double gain;

    @JsonProperty(value="polarization", required = false)
    protected String polarization;


    @JsonProperty(value="cross_polar_discrimination", required = false)
    protected Double crossPolarDiscrimination;

    @JsonProperty(value="horizontal_gain_pattern", required = false)
    protected Double[] horizontalGainPattern;

    @JsonProperty(value="vertical_gain_pattern", required = false)
    protected  Double[] verticalGainPattern;


    @JsonProperty(value="horizontal_beamwidth", required = false)
    protected Double horizontalBeamWidth;


    @JsonProperty(value="vertical_beamwidth", required = false)
    protected Double verticalBeamWidth;


    @JsonProperty(value="voltage_standing_wave_ratio", required = false)
    protected Double voltageStandingWaveRatio;


    @JsonProperty(value="cable_loss", required = false)
    protected Double cableLoss;

    @JsonProperty(value="steerable", required = false)
    protected Boolean steerable;

    @JsonProperty(value="azimuth_angle", required = false)
    protected Double azimuthAngle;


    @JsonProperty(value="elevation_angle", required = false)
    protected Double elevationAngle;

    protected Map<String, Object> otherFields = new HashMap<>();

    public HardwareSpec getAntennaSpec() {
        return antennaSpec;
    }

    public void setAntennaSpec(HardwareSpec antennaSpec) {
        this.antennaSpec = antennaSpec;
    }

    public String getPolarization() {
        return polarization;
    }

    public void setPolarization(String polarization) {
        this.polarization = polarization;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Double getFrequencyLow() {
        return frequencyLow;
    }

    public void setFrequencyLow(Double frequencyLow) {
        this.frequencyLow = frequencyLow;
    }

    public Double getFrequencyHigh() {
        return frequencyHigh;
    }

    public void setFrequencyHigh(Double frequencyHigh) {
        this.frequencyHigh = frequencyHigh;
    }

    public Double getGain() {
        return gain;
    }

    public void setGain(Double gain) {
        this.gain = gain;
    }

    public Double[] getHorizontalGainPattern() {
        return horizontalGainPattern;
    }

    public void setHorizontalGainPattern(Double[] horizontalGainPattern) {
        this.horizontalGainPattern = horizontalGainPattern;
    }

    public Double[] getVerticalGainPattern() {
        return verticalGainPattern;
    }

    public void setVerticalGainPattern(Double[] verticalGainPattern) {
        this.verticalGainPattern = verticalGainPattern;
    }

    public Double getHorizontalBeamWidth() {
        return horizontalBeamWidth;
    }

    public void setHorizontalBeamWidth(Double horizontalBeamWidth) {
        this.horizontalBeamWidth = horizontalBeamWidth;
    }

    public Double getVerticalBeamWidth() {
        return verticalBeamWidth;
    }

    public void setVerticalBeamWidth(Double verticalBeamWidth) {
        this.verticalBeamWidth = verticalBeamWidth;
    }

    public Double getCrossPolarDiscrimination() {
        return crossPolarDiscrimination;
    }

    public void setCrossPolarDiscrimination(Double crossPolarDiscrimination) {
        this.crossPolarDiscrimination = crossPolarDiscrimination;
    }

    public Double getVoltageStandingWaveRatio() {
        return voltageStandingWaveRatio;
    }

    public void setVoltageStandingWaveRatio(Double voltageStandingWaveRatio) {
        this.voltageStandingWaveRatio = voltageStandingWaveRatio;
    }

    public Double getCableLoss() {
        return cableLoss;
    }

    public void setCableLoss(Double cableLoss) {
        this.cableLoss = cableLoss;
    }

    public Boolean getSteerable() {
        return steerable;
    }

    public void setSteerable(Boolean steerable) {
        this.steerable = steerable;
    }

    @JsonAnyGetter
    public Map<String, Object> getOtherFields(){
        return otherFields;
    }

    @JsonAnySetter
    public void add(String key, Object value){
        otherFields.put(key, value);
    }

    public Double getAzimuthAngle() {
        return azimuthAngle;
    }

    public void setAzimuthAngle(Double azimuthAngle) {
        this.azimuthAngle = azimuthAngle;
    }

    public Double getElevationAngle() {
        return elevationAngle;
    }

    public void setElevationAngle(Double elevationAngle) {
        this.elevationAngle = elevationAngle;
    }

}

File: C#/sigmf/sigmf/ext/global/core/Antenna.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf
{
    [Serializable]
    public class Antenna
    {
        private static readonly long serialVersionUID = 1L;

        [JsonProperty("antenna_spec")]
        protected HardwareSpec antennaSpec;

        [JsonProperty("type")]
        protected string type;

        [JsonProperty("low_frequency")]
        protected double lowFrequency;

        [JsonProperty("high_frequency")]
        protected double highFrequency;

        [JsonProperty("gain")]
        protected double gain;

        [JsonProperty("polarization")]
        protected string polarization;

        [JsonProperty("cross_polar_discrimination")]
        protected double crossPolarDiscrimination;

        [JsonProperty("horizontal_gain_pattern")]
        protected List<double> horizontalGainPattern;

        [JsonProperty("vertical_gain_pattern")]
        protected List<double> verticalGainPattern;

        [JsonProperty("horizontal_beam_width")]
        protected double horizontalBeamWidth;

        [JsonProperty("vertical_beam_width")]
        protected double verticalBeamWidth;

        [JsonProperty("voltage_standing_wave_ratio")]
        protected double voltageStandingWaveRatio;

        [JsonProperty("cable_loss")]
        protected double cableLoss;

        [JsonProperty("steerable")]
        protected bool steerable;

        [JsonProperty("azimuth_angle")]
        protected double azimuthAngle;

        [JsonProperty("elevation_angle")]
        protected double elevationAngle;

        [JsonProperty("mobile")]
        protected bool mobile;

        protected Dictionary<string, object> otherfields = new Dictionary<string, object>();

        public HardwareSpec AntennaSpec { get => antennaSpec; set => antennaSpec = value; }
        public string Type { get => type; set => type = value; }
        public double LowFrequency { get => lowFrequency; set => lowFrequency = value; }
        public double HighFrequency { get => highFrequency; set => highFrequency = value; }
        public double Gain { get => gain; set => gain = value; }
        public string Polarization { get => polarization; set => polarization = value; }
        public double CrossPolarDiscrimination { get => crossPolarDiscrimination; set => crossPolarDiscrimination = value; }
        public List<double> HorizontalGainPattern { get => horizontalGainPattern; set => horizontalGainPattern = value; }
        public List<double> VerticalGainPattern { get => verticalGainPattern; set => verticalGainPattern = value; }
        public double HorizontalBeamWidth { get => horizontalBeamWidth; set => horizontalBeamWidth = value; }
        public double VerticalBeamWidth { get => verticalBeamWidth; set => verticalBeamWidth = value; }
        public double VoltageStandingWaveRatio { get => voltageStandingWaveRatio; set => voltageStandingWaveRatio = value; }
        public double CableLoss { get => cableLoss; set => cableLoss = value; }
        public bool Steerable { get => steerable; set => steerable = value; }
        public double AzimuthAngle { get => azimuthAngle; set => azimuthAngle = value; }
        public double ElevationAngle { get => elevationAngle; set => elevationAngle = value; }
        public bool Mobile { get => mobile; set => mobile = value; }

        public Dictionary<string, object> getOtherFields() {
            return otherfields;
        }
        public void add(string key, object value) {
            otherfields.Add(key, value);
        }
    }
}
